docs/introduction-to-smart-contracts.rst

```diff
@@ -115,58 +116,56 @@ registering with username and password — all you need is an Ethereum keypair.
 
-The line ``address public minter;`` declares a state variable of type address
-that is publicly accessible. The ``address`` type is a 160-bit value
-that does not allow any arithmetic operations. It is suitable for
-storing addresses of contracts or of keypairs belonging to external
-persons. The keyword ``public`` automatically generates a function that
-allows you to access the current value of the state variable
-from outside of the contract.
-Without this keyword, other contracts have no way to access the variable.
-The code of the function generated by the compiler is roughly equivalent
+The line ``address public minter;`` declares a state variable of type :ref:`address<address>`.
+The ``address`` type is a 160-bit value that does not allow any arithmetic operations.
+It is suitable for storing addresses of contracts, or a hash of the public half of a keypair belonging to external persons.
+
+The keyword ``public`` automatically generates a function that allows you to access the current value of the state
+variable from outside of the contract. Without this keyword, other contracts have no way to access the variable.
+The code of the function generated by the compiler is equivalent
 to the following (ignore ``external`` and ``view`` for now)::
 
     function minter() external view returns (address) { return minter; }
 
-Of course, adding a function exactly like that will not work
-because we would have a
-function and a state variable with the same name, but hopefully, you
-get the idea - the compiler figures that out for you.
+You could add a function like the above yourself, but you would have a function and state variable with the same name.
+You do not need to do this, the compiler figures it out for you.
 
 .. index:: mapping
 
 The next line, ``mapping (address => uint) public balances;`` also
 creates a public state variable, but it is a more complex datatype.
-The type maps addresses to unsigned integers.
+The :ref:`mapping <mapping-types>` type maps addresses to :ref:`unsigned integers <integers>`.
+
 Mappings can be seen as `hash tables <https://en.wikipedia.org/wiki/Hash_table>`_ which are
-virtually initialized such that every possible key exists from the start and is mapped to a
-value whose byte-representation is all zeros. This analogy does not go
-too far, though, as it is neither possible to obtain a list of all keys of
-a mapping, nor a list of all values. So either keep in mind (or
-better, keep a list or use a more advanced data type) what you
-added to the mapping or use it in a context where this is not needed.
+virtually initialised such that every possible key exists from the start and is mapped to a
+value whose byte-representation is all zeros. However, it is neither possible to obtain a list of all keys of
+a mapping, nor a list of all values. Keep in mind what you
+added to the mapping or use it in a context where this is not needed. Or
+even better, keep a list, or use a more suitable data type.
```

Comment:
And then this can be deleted

Reply:
Which part? I feel like at least a part of this explanation is still useful.

Reply:
The sentence "Or even better, keep a list, ..."

Reply:
Actually it is fine as it is.